lib/ShareByCircleProvider.php

```diff
@@ -637,10 +662,84 @@ public function getAccessList($nodes, $currentAccess): array {
 			$ids[] = $node->getId();
 		}
 
-		$users = $remote = $mails = [];
-		$shares = $this->shareWrapperService->getSharesByFileIds($ids, true);
 
-		foreach($shares as $share) {
+		if (!$currentAccess) {
+			return $this->getAccessListShort($ids);
+		}
+
+		$shareIds = $knownIds = $users = $remote = $mails = [];
+		foreach ($this->shareWrapperService->getSharesByFileIds($ids, true, true) as $share) {
+			$shareIds[] = $share->getId();
+			$circle = $share->getCircle();
+			foreach ($circle->getInheritedMembers() as $member) {
+				if ($share->getParent() > 0 && in_array($member->getSingleId(), $knownIds[$share->getFileSource()] ?? [])) {
+					continue;
+				}
+				$knownIds[$share->getFileSource()][] = $member->getSingleId();
+
+				switch ($member->getUserType()) {
+					case Member::TYPE_USER:
+						if ($member->isLocal()) {
+							$users[$member->getUserId()] = [
+								'node_id' => $share->getFileSource(),
+								'node_path' => $share->getFileTarget()
+							];
+						} else {
+							// we only store temp value, as token is unknown at this point
+							$remote[$member->getUserid() . '@' . $member->getInstance()] = [
+								'node_id' => $share->getFileSource(),
+								'shareId' => $share->getId(),
+								'memberId' => $member->getMemberId(),
```

Comment:
```suggestion
								'memberId' => $member->getId(),
```

Maybe? I have no clue, apart from that getMemberId indeed does not exist.